### DISCUSSION TITLE
How can I use allauth in API?

### DISCUSSION THREAD

**Discussion by LucasLeone** [cookiecutter/cookiecutter-django#4992] [Q&A]
Hi everybody!

How can I use allauth in API? I want to do a REST API so I don't want to use the users' module.

Thanks!

**Comment by browniebroke** [comment]
I've used https://django-rest-auth.readthedocs.io in a project, and it integrates with allauth, however I later heard some good things about https://jazzband.github.io/django-rest-knox to do API auth well. I have no idea if that second option integrates with allauth though.

**Reply by foarsitter** [reply]
The author of allauth has received a grand to achieve API support: https://allauth.org/news/2024/03/ngi-zero-grant-plan/ but don't know anything about the progress.

Update: almost there I see: https://allauth.org/news/2024/04/api-feedback/

**Reply by browniebroke** [reply]
Looks like it has been released in 0.63: https://docs.allauth.org/en/latest/release-notes/recent.html

> An official API for single-page and mobile application support is now available, via the new `allauth.headless` app.

Documentation: https://docs.allauth.org/en/latest/headless/index.html

**Reply by foarsitter** [reply]
Had a short discussion with the author and I'm struggling with the implementation. It is designed as an separated API and I'm not very fond of it to supply a second API to my clients: https://github.com/pennersr/django-allauth/discussions/3737

**Comment by quroom** [comment]
I more like https://dj-rest-auth.readthedocs.io/en/latest/
If you are considering jwt token authentication and refresh token in frontned with backend.
It's easier to use `dj-rest-auth` beacuse django-rest-auth doesn't support jwt refresh endpoint

**Comment by LucasLeone** [comment]
I'm using TokenAuthentication, maybe I'm gonna wait for allauth API support. Thanks everybody!

**Comment by foarsitter** [comment]
The API allauth is developing will be a standalone API without any DRF integration. So I don't think it will benefit us since we cannot incorporate the endpoints in the rest of the API. So you will point your clients to a different OpenAPI spec when it comes about authentication.
